src/prpy/serialization.py

```diff
@@ -99,43 +101,60 @@ def serialize_environment_file(env, path, writer=None):
 
     return data
 
-def serialize_kinbody(body):
-    all_joints = []
-    all_joints.extend(body.GetJoints())
-    all_joints.extend(body.GetPassiveJoints())
+def serialize_kinbody(body, uri_only=False):
 
+    uri = body.GetXMLFilename()
+    
+    if not uri:
+        uri_only = False
+    
     data = {
         'is_robot': body.IsRobot(),
         'name': body.GetName(),
-        'uri': body.GetXMLFilename(),
-        'links': map(serialize_link, body.GetLinks()),
-        'joints': map(serialize_joint, all_joints),
+        'uri': uri,
     }
-    data['kinbody_state'] = serialize_kinbody_state(body)
+    
+    if uri_only:
+        data['uri_only'] = True
+    
+    if not uri_only:
+        all_joints = []
+        all_joints.extend(body.GetJoints())
+        all_joints.extend(body.GetPassiveJoints())
+        data['links'] = map(serialize_link, body.GetLinks())
+        data['joints'] = map(serialize_joint, all_joints)
+    
+    data['kinbody_state'] = serialize_kinbody_state(body, uri_only=uri_only)
 
     if body.IsRobot():
-        data.update(serialize_robot(body))
+        data.update(serialize_robot(body, uri_only=uri_only))
 
     return data
 
-def serialize_robot(robot):
-    return {
-        'manipulators': map(serialize_manipulator, robot.GetManipulators()),
+def serialize_robot(robot, uri_only=False):
+    data = {
         'robot_state': serialize_robot_state(robot),
     }
+    if not uri_only:
+        data.update({
+            'manipulators': map(serialize_manipulator, robot.GetManipulators()),
+        })
+    return data
 
-def serialize_kinbody_state(body):
+def serialize_kinbody_state(body, uri_only=False):
     data = {
         name: get_fn(body)
         for name, (get_fn, _) in KINBODY_STATE_MAP.iteritems()
     }
 
     link_transforms, dof_branches = body.GetLinkTransformations(True)
     data.update({
-        'link_transforms': map(serialize_transform, link_transforms),
-        'dof_branches': dof_branches.tolist(),
```

Comment:
Fixed! They were removed because I originally wanted to replicate just loading the robot and setting DOF values.